### PR TITLE
docs: EXPOSED-445 Add documentation for DSL & DAO composite primary keys

### PR DESCRIPTION
**Pull request by bog-walk**
Adds details about new `CompositeIdTable` and how it relates to:

- Available `IdTable` types
- DAO CRUD operations
- DAO references
- Creating primary keys
- Creating foreign keys

Additional:
- Update incorrect FAQ about ARRAY column type
- Replace FAQ about custom `IdTable` with a direct example
- Restructure `hi.tree` to include existing, but unused, `Table-Definitions.md`

### CODE REVIEW

File: documentation-website/Writerside/hi.tree [hunk 1]
```diff
@@ -8,14 +8,15 @@
...
-    <toc-element topic="Deep-Dive-into-DSL.md"/>
-    <toc-element topic="Deep-Dive-into-DAO.md"/>
-    <toc-element topic="Exposed-Modules.md">
+    <toc-element topic="Table-Definition.md">
         <toc-element topic="Data-Types.md"/>
         <toc-element topic="SQL-Functions.md"/>
     </toc-element>
+    <toc-element topic="Transactions.md"/>
+    <toc-element topic="Deep-Dive-into-DSL.md"/>
+    <toc-element topic="Deep-Dive-into-DAO.md"/>
```
Comment (bog-walk) [inline documentation-website/Writerside/hi.tree:19]: @vnikolova This may not match exactly with the proposed structure on the drive doc, so let me know if I should change.

`Table-Definition.md` was not being used despite all its contents, so I've added it and used it as the parent of `Data-Types` and `SQL-Functions`, which makes more sense than the file with modules and dependencies.
I've also moved these to follow the content on databases, since schema definition seems to usually come after a connection is set up, but before a transaction is used.

**Review by e5l — APPROVED** [review]

**Review by obabichevjb — APPROVED** [review]

File: documentation-website/Writerside/topics/Table-Definition.md [hunk 2]
```diff
@@ -146,42 +194,63 @@ val name = varchar("name", 50).uniqueIndex()
...
+* `onUpdate: ReferenceOption? = null` is an action for when a value in a referenced column will be changed.
+* `fkName: String? = null` is a name for the foreign key constraint.
 
-Enum class `ReferenceOption` has four values:
+Enum class `ReferenceOption` has five values:
 
-* `RESTRICT` is an option that restricts changes on a referenced column, and the default option for MySQL dialect.
-* `NO_ACTION` is the same as RESTRICT, and the default option for Oracle and SQL Server dialects.
+* `RESTRICT` is an option that restricts changes on a referenced column, and the default option for most dialects.
+* `NO_ACTION` is the same as RESTRICT in some, but not all, databases, and the default option for Oracle and SQL Server dialects.
```
Comment (vnikolova) [inline documentation-website/Writerside/topics/Table-Definition.md:247]: Here and in any other places where we have lists of options/properties, I think it would be nice to start using the [`<deflist>`](https://www.jetbrains.com/help/writerside/lists.html#type-wide) element from Writerside. It can be in a separate PR though.

File: documentation-website/Writerside/topics/Deep-Dive-into-DAO.md [hunk 3]
```diff
@@ -282,6 +367,65 @@ val child2 = Node.new { name = "child2" }
...
+    companion object : CompositeEntityClass<Director>(Directors)
+    var genre by Directors.genre
+    val films by StarWarsFilm referrersOn StarWarsFilms
+}
+```
+You can then access this field on a `Director` object, `director`:
+```kotlin
+director.films // returns all StarWarsFilm objects that reference this director
+```
+Using other previously mentioned infix functions, like `optionalReferencedOn`, `backReferencedOn`, and `optionalReferrersOn`,
```
Comment (vnikolova) [inline documentation-website/Writerside/topics/Deep-Dive-into-DAO.md:425]: What is an `infix` function? I see an `infix` call also mentioned on line 287 but I can't find the definition anywhere within the docs. Maybe we could add a link?

File: documentation-website/Writerside/hi.tree (hunk 1 above)
Comment (vnikolova) [reply documentation-website/Writerside/hi.tree:19]: I'm not sure if `SQL-Functions` makes sense to be under `Table-Definitions` because it's not part of defining a Table, right? Perhaps we could have an empty `toc-element` such as "Working with Tables" that will be the parent of `Table-Definition`, `Data-Types` and `SQL-Functions`. What do you think?

File: documentation-website/Writerside/topics/Table-Definition.md (hunk 2 above)
Comment (bog-walk) [reply documentation-website/Writerside/topics/Table-Definition.md:247]: Sure, I will at least use it in the places I have changes already.

Comment (vnikolova) [reply documentation-website/Writerside/topics/Table-Definition.md]: That's ok. It also makes sense as the file is already in Markdown. I think in XML you'd have to use the `<code>` element in the title.

**Review by vnikolova — APPROVED** [review]
